plutus-core/executables/uplc/Main.hs

```diff
@@ -218,6 +244,54 @@ runEval (EvalOptions inp ifmt printMode budgetMode traceMode outputMode timingMo
                         None -> pure ()
                         _    -> writeToFileOrStd outputMode (T.unpack (T.intercalate "\n" logs))
 
+---------------- Debugging ----------------
+
+runDbg :: DbgOptions -> IO ()
+runDbg (DbgOptions inp ifmt cekModel) = do
+    prog <- getProgram ifmt inp
+    let term = prog ^. UPLC.progTerm
+        !_ = rnf term
+        nterm = fromRight (error "Term to debug must be closed.") $
+                   runExcept @FreeVariableError $ deBruijnTerm term
+    let cekparams = case cekModel of
+                    -- AST nodes are charged according to the default cost model
+                    Default -> PLC.defaultCekParameters
+                    -- AST nodes are charged one unit each, so we can see how many times each node
+                    -- type is encountered.  This is useful for calibrating the budgeting code
+                    Unit    -> PLC.unitCekParameters
+        MachineParameters costs runtime = cekparams
+    let ?cekRuntime = runtime
+        ?cekEmitter = const $ pure ()
+        ?cekBudgetSpender = Cek.CekBudgetSpender $ \_ _ -> pure ()
```

Comment:
So currently the debug CEK machine does not keep track of the cost. Which is fine, but could you make a separate ticket for that since it will be needed.